modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/snapshot/local/LocalSnapshotCopier.java

```diff
@@ -260,6 +260,10 @@ private void loadMetaTable() throws InterruptedException {
                 setError(session.status().getCode(), session.status().getErrorMsg());
                 return;
             }
+            // No need in further snapshot processing if it was already cancelled and the metaBuf is null.
+            if (metaBuf.getBuffer() == null && session.status().isCanceled()) {
```

Comment:
Why don't you check `&& isOk()` in the condition like above before setting another state?

Reply:
What do you mean? The intent is to prevent further processing is metaBuf is null and request was canceled.